ask-sdk-servlet-support/src/com/amazon/ask/servlet/verifiers/GenericTimestampRequest.java

```diff
@@ -0,0 +1,114 @@
+package com.amazon.ask.servlet.verifiers;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import java.time.OffsetDateTime;
+import java.util.Objects;
+
+@JsonDeserialize(
+        builder = GenericTimestampRequest.Builder.class
+)
+public class GenericTimestampRequest {
```

Comment:
I feel this belongs in the model package. We'd have to create some sort of base class and have RequestEnvelope inherit and extend .

Reply:
the problem is that when you deal with requests that are not present in the model package. I think that request validation should not rely on the model package - you should have anemic request class that should deserialize to fields required only by validation logic

Reply:
I agree to some degree, I think this function should only extract what it needs.  But that should be defined as a generic object in the model. Otherwise, we have two packages where a request is defined.   i.e  however unlikely, if the timestamp field moves/renamed, this would be hard to catch in the future. 

so not entirely against this just need to think about this a bit more. Currently I see more pros in having this in the model than cons.

Reply:
the biggest problem here is that I cannot edit your template files since they're private. Also having a base class can be tricky when relying on builders to create objects. 

I know that there is a defaultImpl property in Jackson: https://fasterxml.github.io/jackson-annotations/javadoc/2.6/com/fasterxml/jackson/annotation/JsonTypeInfo.html#defaultImpl()

So a request could be mapped to `UnknownRequest` that would have locale, type, request-id, and timestamp.